components/table/nz-table.component.ts

```diff
@@ -213,21 +220,24 @@ export class NzTableComponent<T = any> implements OnInit, AfterViewInit, OnDestr
   }
 
   updateFrontPaginationDataIfNeeded(isPageSizeOrDataChange: boolean = false): void {
-    let data = this.nzData || [];
     if (this.nzFrontPagination) {
-      this.nzTotal = data.length;
+      this.nzTotal = this.data.length;
```

Comment:
This is not ideal, as the NzDataSource instance might not have the value available at this point. I need to refactor this, somehow, into an "asynchronous" process

Reply:
agree, let's do this :)